specs/eip4844/fork-choice.md

```diff
@@ -23,54 +22,42 @@ This is the modification of the fork choice accompanying the EIP-4844 upgrade.
 
 ## Containers
 
-### `BlobsSidecar`
-
-```python
-class BlobsSidecar(Container):
-    beacon_block_root: Root
-    beacon_block_slot: Slot
-    blobs: List[Blob, MAX_BLOBS_PER_BLOCK]
-    kzg_aggregated_proof: KZGProof
-```
-
 ## Helpers
 
-#### `validate_blobs_sidecar`
+#### `validate_blob_sidecars`
 
 ```python
-def validate_blobs_sidecar(slot: Slot,
+def validate_blob_sidecars(slot: Slot,
                            beacon_block_root: Root,
                            expected_kzg_commitments: Sequence[KZGCommitment],
-                           blobs_sidecar: BlobsSidecar) -> None:
+                           blob_sidecars: Sequence[BlobSidecar]) -> None:
     assert slot == blobs_sidecar.beacon_block_slot
     assert beacon_block_root == blobs_sidecar.beacon_block_root
-    blobs = blobs_sidecar.blobs
-    kzg_aggregated_proof = blobs_sidecar.kzg_aggregated_proof
-    assert len(expected_kzg_commitments) == len(blobs)
-
-    assert verify_aggregate_kzg_proof(blobs, expected_kzg_commitments, kzg_aggregated_proof)
+    assert len(expected_kzg_commitments) == len(blob_sidecars)
+    # TODO validate commitments individually or aggregate first?
```

Comment:
There is no `BlobsSidecar` that contained `kzg_aggregated_proof` so aggregate check just isn't possible here. I suggest getting rid of `kzg_aggregated_proof` completely and everywhere simply use individual commitments instead.

Reply:
This is what we're currently planning to do

Reply:
> I suggest getting rid of kzg_aggregated_proof completely and everywhere simply use individual commitments instead.

My understanding was that as a client, it might be more efficient to aggregate-then-validate, iff I have multiple proofs to check?

Reply:
The crypto layer will be [providing a function](https://github.com/ethereum/consensus-specs/pull/3236/files#diff-44e74c6479198ca2d57ab8c0895785dc5ab016145e9472de9bdcd94ab105265bR510) that can verify multiple blobs at the same time. It will **not use** an aggregated proof to do so.

It will still be faster than verifying the individual proofs sequentially, but slower than using an aggregated proof.

Reply:
da34af97d4b6575131e67c2bf22acb6de4f0951d